### ISSUE TITLE
Add support for configuring Kubernetes startup probe

### ISSUE THREAD

**Issue by yersan** [wildfly/wildfly-charts#37]
## Overview

The probes (Liveness and Readiness) added by the Helm Chart uses the following default values:

```
failureThreshold: 3
periodSeconds: 10
successThreshold: 1
timeoutSeconds: 1
initialDelaySeconds: 0
```

For the liveness probes, for example, this configuration means the server could take until `failureThreshold*periodSeconds=30 seconds` to get started. If such a condition doesn't occur the cluster will kill the pod to start over it again (restart policy is always).

Sometimes the pod could need more than 30 seconds to start, for example:
- The user has registered other liveness checks in their application code and to satisfy those conditions is needed more time.
- The management operations executed at runtime when the pod is started could also take some time to be applied, which delays the availability of the endpoint used by the default liveness probe.
- The node where the pod is running is slow.

In such a case, users could be tempted to increase the `initialDelaySeconds` or the failure `failureThreshold` configuration, however increasing the `initialDelaySeconds` will delay a fixed value, which could compromise the expected fast response. Increasing the `failureThreshold` could also break the established conditions you have decided to check your application at runtime.

The alternative to those issues is to have the ability to configure a [Startup Probe](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-startup-probes). The Startup Probe can be used to delay the initial checks done by the liveness and readiness probes, allowing some configuration without compromising them.

The goal of this RFE is to define the default initial configuration of a Startup Probe so users can tweak it when required.

## Dev Contacts
Yeray Borgess yborgess@redhat.com
## QE Contacts
TBD
## Testing By
TBD
## Affected Projects or Components
[WildFly Charts](https://docs.wildfly.org/wildfly-charts/)
## Other Interested Projects
N/A
### Hard Requirements
- By default, the Chart will include a Startup Probe to give three minutes as additional time before starting the Readiness and Liveness probes.
- The proposal for the initial values is the following:
```
failureThreshold: 36
periodSeconds: 5
successThreshold: 1
timeoutSeconds: 1
initialDelaySeconds: 5
```
- Users should be able to tweak any of those default values when they are installing the chart.

### Nice-to-Have Requirements
N/A

### Non-Requirements
N/A

## Test Plan
- Manual verification of the expected values.

## Community Documentation
The Helm Chart docs will include a description of the uses of the Startup Probe.

## Release Note Content
Added the ability to configure a Startup Probe.
